### ISSUE TITLE
`pythia8-config` is broken for pythia8 v8.311

### ISSUE THREAD

**Issue by matthewfeickert** [conda-forge/pythia8-feedstock#45]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

```console
$ micromamba env create --yes --name test pythia8
$ micromamba activate test
$ micromamba list | grep pythia8
  pythia8           8.311     py312hca68cad_0     conda-forge
$ command -v pythia8-config
/root/micromamba/envs/test/bin/pythia8-config
$ pythia8-config --prefix
Error: cannot find valid configuration for Pythia 8
$ pythia8-config --cxxflags
Error: cannot find valid configuration for Pythia 8
```

but `pythia8` `v8.310` is not affected:

```console
$ micromamba deactivate 
$ micromamba env create --yes --name test-8310 pythia8=8.310
$ micromamba activate test-8310
$ micromamba list | grep pythia8
  pythia8           8.310     py312hca68cad_2     conda-forge
$ pythia8-config --prefix
/root/micromamba/envs/test-8310
$ pythia8-config --cxxflags
-I/root/micromamba/envs/test-8310/include
```

### Installed packages

```shell
Name              Version   Build               Channel    
───────────────────────────────────────────────────────────────
  _libgcc_mutex     0.1       conda_forge         conda-forge
  _openmp_mutex     4.5       2_gnu               conda-forge
  bzip2             1.0.8     h4bc722e_7          conda-forge
  ca-certificates   2024.7.4  hbcca054_0          conda-forge
  ld_impl_linux-64  2.40      hf3520f5_7          conda-forge
  libexpat          2.6.2     h59595ed_0          conda-forge
  libffi            3.4.2     h7f98852_5          conda-forge
  libgcc-ng         14.1.0    h77fa898_0          conda-forge
  libgomp           14.1.0    h77fa898_0          conda-forge
  libnsl            2.0.1     hd590300_0          conda-forge
  libsqlite         3.46.0    hde9e2c9_0          conda-forge
  libstdcxx-ng      14.1.0    hc0a3c3a_0          conda-forge
  libuuid           2.38.1    h0b41bf4_0          conda-forge
  libxcrypt         4.4.36    hd590300_1          conda-forge
  libzlib           1.3.1     h4ab18f5_1          conda-forge
  ncurses           6.5       h59595ed_0          conda-forge
  openssl           3.3.1     h4bc722e_2          conda-forge
  pip               24.2      pyhd8ed1ab_0        conda-forge
  pythia8           8.311     py312hca68cad_0     conda-forge
  python            3.12.5    h2ad013b_0_cpython  conda-forge
  python_abi        3.12      4_cp312             conda-forge
  readline          8.2       h8228510_1          conda-forge
  setuptools        72.1.0    pyhd8ed1ab_0        conda-forge
  tk                8.6.13    noxft_h4845f30_101  conda-forge
  tzdata            2024a     h0c530f3_0          conda-forge
  wheel             0.44.0    pyhd8ed1ab_0        conda-forge
  xz                5.2.6     h166bdaf_0          conda-forge
```

### Environment info

```shell
libmamba version : 1.5.8
     micromamba version : 1.5.8
           curl version : libcurl/8.6.0 OpenSSL/3.2.1 zlib/1.2.13 zstd/1.5.5 libssh2/1.11.0 nghttp2/1.58.0
     libarchive version : libarchive 3.7.2 zlib/1.2.13 bz2lib/1.0.8 libzstd/1.5.5
       envs directories : /root/micromamba/envs
          package cache : /root/micromamba/pkgs
                          /root/.mamba/pkgs
            environment : test (active)
           env location : /root/micromamba/envs/test
      user config files : /root/.mambarc
 populated config files : /root/.condarc
       virtual packages : __unix=0=0
                          __linux=6.5.0=0
                          __glibc=2.36=0
                          __archspec=1=x86_64-v3
               channels : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/nodefaults/linux-64
                          https://conda.anaconda.org/nodefaults/noarch
       base environment : /root/micromamba
               platform : linux-64
```

**Comment by matthewfeickert** [comment]
So far what I've been able to identify on a working `pythia` `v8.310` build is that the `CFG_FILE` in the `pythia8-config` shell script is getting properly set to the environment's `share/Pythia8/examples/Makefile.inc`

```console
# grep "CFG_FILE=" $(command -v pythia8-config)
CFG_FILE=/root/micromamba/envs/test-8310/share/Pythia8/examples/Makefile.inc
if [ -f $PREFIX/Makefile.inc ]; then CFG_FILE="$PREFIX/Makefile.inc"
```

while the failing `pythia` `v8.311` build has 

```console
# grep "CFG_FILE=" $(command -v pythia8-config)
CFG_FILE=examples/Makefile.inc
if [ -f $PREFIX/Makefile.inc ]; then CFG_FILE="$PREFIX/Makefile.inc"
```

**Comment by matthewfeickert** [comment]
There's a bug somewhere in the Pythia v8.311 source code (that will need to get patched) as if I use the same branch and then just apply the following diff to switch from `v8.311` to `v8.310`

```diff
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index f543b3d..bca890c 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,5 +1,5 @@
 {% set name = "pythia8" %}
-{% set file_version = "8311" %}
+{% set file_version = "8310" %}
 {% set version = file_version[0] + "." + file_version[1:] %}
 
 package:
@@ -8,7 +8,7 @@ package:
 
 source:
   url: https://gitlab.com/Pythia8/releases/-/archive/pythia{{ file_version }}/releases-pythia{{ file_version }}.tar.gz
-  sha256: 6cca3920a89a5e51d259114804f3f4944133c1d3883d3f454162ccd3f3bd3c4d
+  sha256: 00cb6c74b3f530fafc297d515c3e3e037593dd1867cdd0b1ee2d481251cfec35
 
 build:
   number: 1
```

and run

```
python build-locally.py linux_64_python3.12.____cpython
```

the resulting `v8.310` conda package works when installed locally

```
# cd build_artifacts
micromamba install --channel ./linux-64 pythia8=8.310
```

**Comment by matthewfeickert** [comment]
It seems that Pythia `8.311` is a broken release as using the same dev branch but switching up to `8.312`

```diff
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index bca890c..153fdc5 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,5 +1,5 @@
 {% set name = "pythia8" %}
-{% set file_version = "8310" %}
+{% set file_version = "8312" %}
 {% set version = file_version[0] + "." + file_version[1:] %}
 
 package:
@@ -8,7 +8,7 @@ package:
 
 source:
   url: https://gitlab.com/Pythia8/releases/-/archive/pythia{{ file_version }}/releases-pythia{{ file_version }}.tar.gz
-  sha256: 00cb6c74b3f530fafc297d515c3e3e037593dd1867cdd0b1ee2d481251cfec35
+  sha256: c1a33aa5fa15e6b70d7946ce6d237246842887ec84ea0b35dfc2535c868a2770
 
 build:
   number: 1
```

and running

```
python build-locally.py linux_64_python3.12.____cpython
```

the resulting `v8.312` conda package works when installed locally

```
# cd build_artifacts
micromamba install --channel ./linux-64 pythia8=8.312
```

```console
$ pythia8-config --prefix
/root/micromamba/envs/pythia8-8312-local-build-test
```

So [`8.310`](https://gitlab.com/Pythia8/releases/-/tree/pythia8310) and [`8.312`](https://gitlab.com/Pythia8/releases/-/tree/pythia8312) build and install fine, but [`8.311`](https://gitlab.com/Pythia8/releases/-/tree/pythia8311) is currently broken and will need a patch for the conda-forge distribution to work.

**Comment by matthewfeickert** [comment]
@ph-ilten, as you're a PYTHIA expert's expert :) do you have any thoughts or knowledge on what the change here was that entered into `8.311` and was patched in `8.312` so that I can apply a similar patch here?

Looking at the [PYTHIA8 update history](https://pythia.org/history/) I see under "8.312: 23 May 2024"

> Tiny correction in configure that fixes issue with pythia8-config on some platforms, thanks to Gabriele Inghirami for reporting.

Do you know what patch this was? (Apparently one of the changes in the following diff between tags, and I'm guessing the only line that contains `CFG_FILE`?)

```diff
$ git diff pythia8311 pythia8312 -- configure 
diff --git a/configure b/configure
index e2d35cfc..d7fd367d 100755
--- a/configure
+++ b/configure
@@ -339,6 +339,12 @@ if [ -z "$LIB_SUFFIX" ]; then
     if [ "$ARCH" = "LINUX" ];  then LIB_SUFFIX=".so"; fi
     if [ "$ARCH" = "DARWIN" ]; then LIB_SUFFIX=".dylib"; fi; fi
 
+# Handle new versus old dynamic tags (prefer old transitive behaviour).
+CXX_DTAGS=$(echo "int main(){}" | $CXX -Wl,--disable-new-dtags -o \
+    /dev/null -xc - 2>&1)
+if [ -z "$CXX_DTAGS" ]; then CXX_DTAGS="-Wl,--disable-new-dtags";
+else CXX_DTAGS=""; fi
+
 # Configure the installation directories (order matters).
 if [ "$PREFIX_SET" != true ]; then PREFIX=$PWD; fi
 if [ -z "$PREFIX_BIN" ];      then PREFIX_BIN=$PREFIX/bin; fi
@@ -383,7 +389,7 @@ configure_package "MPICH" "" "" "" "" "" "" "mpic++" "mpi.h" "mpi" ""
 configure_package "HDF5" "$LCG_PATH/hdf5" "1.12.2" "" "" "" "" "" "hdf5.h"\
     "hdf5" ""
 configure_package "HIGHFIVE" "$LCG_PATH/highfive" "2.0" "" "" "" "" ""\
-    "H5File.hpp" "" ""
+    "highfive/H5File.hpp" "" ""
 
 # Check package dependencies (order does not matter).
 for IDX in ${!PKG_LIST[@]}; do
@@ -428,7 +434,7 @@ done
     
 # Write the configuation to screen and file, update the pythia8-config script.
 mkdir -p bin;
-sed 's|^CFG_FILE=.*|CFG_FILE='"$PREFIX_SHARE"'/examples/Makefile.inc|;n' \
+sed 's|^CFG_FILE=.*|CFG_FILE='"$PREFIX_SHARE"'/examples/Makefile.inc|' \
     pythia8-config.inc > bin/pythia8-config
 chmod a+x bin/pythia8-config
 echo "---------------------------------------------------------------------"
@@ -436,6 +442,7 @@ echo "|                    PYTHIA Configuration Summary                   |"
 echo "---------------------------------------------------------------------"
 echo "  Architecture                = "$ARCH 
 echo "  C++ compiler     CXX        = "$CXX
+echo "  C++ dynamic tags CXX_DTAGS  = "$CXX_DTAGS
 echo "  C++ flags        CXX_COMMON = "$CXX_COMMON
 echo "  C++ shared flag  CXX_SHARED = "$CXX_SHARED
 echo -n "  Further options             = "
@@ -454,6 +461,7 @@ PREFIX_SHARE=$PREFIX_SHARE
 
 # Compilation flags (see ./configure --help for further documentation).
 CXX=$CXX
+CXX_DTAGS=$CXX_DTAGS
 CXX_COMMON=$CXX_COMMON
 CXX_SHARED=$CXX_SHARED
 CXX_SONAME=$CXX_SONAME
```

**Comment by matthewfeickert** [comment]
> (Apparently one of the changes in the following diff between tags, and I'm guessing the only line that contains `CFG_FILE`?)

Yeah, applying the following patch

```console
$ cat 0001-fix-cfg_file-setting-in-configure-8.311.patch 
From 21247059a6d5aa78facf6655009af949b1864924 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Thu, 15 Aug 2024 03:05:36 -0500
Subject: [PATCH] fix-cfg_file-setting-in-configure-8.311

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index e2d35cfc..d052a4a2 100755
--- a/configure
+++ b/configure
@@ -428,7 +428,7 @@ done
     
 # Write the configuation to screen and file, update the pythia8-config script.
 mkdir -p bin;
-sed 's|^CFG_FILE=.*|CFG_FILE='"$PREFIX_SHARE"'/examples/Makefile.inc|;n' \
+sed 's|^CFG_FILE=.*|CFG_FILE='"$PREFIX_SHARE"'/examples/Makefile.inc|' \
     pythia8-config.inc > bin/pythia8-config
 chmod a+x bin/pythia8-config
 echo "---------------------------------------------------------------------"
-- 
2.46.0

```

gets 8.311 building correctly. :+1:

**Comment by ph-ilten** [comment]
> > (Apparently one of the changes in the following diff between tags, and I'm guessing the only line that contains `CFG_FILE`?)
> 
> Yeah, applying the following patch

Yes, this is the patch that makes the difference for some version of `sed`.
